include/extractor/travel_mode.hpp

```diff
@@ -53,4 +55,7 @@ const constexpr osrm::extractor::TravelMode TRAVEL_MODE_RIVER_UP = 10;
 const constexpr osrm::extractor::TravelMode TRAVEL_MODE_RIVER_DOWN = 11;
 const constexpr osrm::extractor::TravelMode TRAVEL_MODE_ROUTE = 12;
 
+// travel modes for which navigation should be suppressed
+const std::array<osrm::extractor::TravelMode, 2> constexpr SUPPRESS_MODE_LIST = {{TRAVEL_MODE_TRAIN, TRAVEL_MODE_FERRY}};
```

Comment:
The problem with using std::array here is the size has to be known at compile time. If you now add an additional mode you have to go through the code base and change all array<T, N> to array<T, N + 1>.